notion-appimage: init at version 4.23.0

jcorbalanm opened the pull request:
Created a new package that wraps a Notion AppImage found here: https://github.com/jcorbalanm/notion-appimage. Set myself as maintainer.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

kashw2 commented:
Is there anything that isn't covered by the current [notion-app](https://github.com/NixOS/nixpkgs/blob/nixos-25.05/pkgs/by-name/no/notion-app) package here?

dotlambda reviewed (dismissed):
Add Linux support to `notion-app` instead and follow https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions.

jcorbalanm commented:
> Add Linux support to `notion-app` instead and follow github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions.

Notion itself does not provide a Linux binary. Repackaging for Linux is above my capabilities.

Updated an squashed to comply with commit conventions.

dotlambda commented on pkgs/by-name/no/notion-appimage/package.nix:
```diff
@@ -0,0 +1,35 @@
+    hash = "sha256:20c8a387bdb71c1ae41a0adf570cda0e9fda3f1dea4e91b5589239e5949ab21b";
+  };
+
+  appimageContents = appimageTools.extract { inherit pname version src; };
```
This isn't used anywhere.

dotlambda commented on pkgs/by-name/no/notion-appimage/package.nix:
```diff
@@ -0,0 +1,35 @@
+
+  src = fetchurl {
+    url = "https://github.com/jcorbalanm/notion-appimage/releases/download/${version}/Notion-${version}-x86_64.AppImage";
+    hash = "sha256:20c8a387bdb71c1ae41a0adf570cda0e9fda3f1dea4e91b5589239e5949ab21b";
```
Use an SRI hash.

acid-bong replied:
(which is generated with emptying the `hash` string and trying to build `PACKAGE.src`)

jcorbalanm commented:
> > Updated an squashed to comply with commit conventions.
> 
> There should be a separate commit `maintainers: add jcorbalanm`.
> 
> > Notion itself does not provide a Linux binary. Repackaging for Linux is above my capabilities.
> 
> Can't we use [jcorbalanm/notion-appimage@`main`/build.sh](https://github.com/jcorbalanm/notion-appimage/blob/main/build.sh?rgh-link-date=2025-11-11T22%3A14%3A04Z) to do the repackaging?

We could do that, along with https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=notion-app-electron shouldn't be too hard, but i have never packaged anything and seemed a bit steep for me. I'm sorry if I am complicating things with an AppImage.

jcorbalanm committed:
notion-appimage: init at version 4.23.0

Wraps a Notion AppImage found in: github.com/jcorbalanm/notion-appimage. Motivation: Lack of NixOS package for Notion. Link to release notes.

jcorbalanm committed:
maintainers: add jcorbalan

jcorbalanm commented:
> 1. title conventions apply to both commits and PRs. you got the commits right, only rename the PR
> 
>     2. maintainer addition must precede new content. may `git rebase -i` be with you

Done, thanks

dotlambda commented:
@qzylinra I've never heard of a Nixpkgs policy like that. In fact, I'm sure I'm the maintainer of multiple packages with zero users, and more with at most one user.

qzylinra commented:
please see https://github.com/NixOS/nixpkgs/pull/404932

dotlambda commented:
> Anyway, I won't be the one merging this unless it's built from an official Notion app rather than a repackaged AppImage.

Did you read what I wrote??

qzylinra commented:
I know, but what you're saying isn't reality, the author hasn't accepted these ideas yet.

jcorbalanm commented:
I understand closing the PR and not wanting to add an AppImage as package. Maybe in a few months I will try to make a native version in a new pull request, to keep everything cleaner. I don't have the time as of right now. Thanks to everyone for your feedback.

Sorry for late reply, I have had a few busy days.
